Stats: Add new navigation

psealock opened the pull request:
Enhance Stats Navigation to make it flexible enough to include more navigation items. 
* Move nav items dealing with intervals (day/week/month/year) into a segmented-control component which can be shown/hidden if applicable. 
* Keep nav items clear:  "Traffic", "Insights", "Activity", "Store".

### Before
<img width="930" alt="screen shot 2017-09-26 at 3 09 13 pm" src="https://user-images.githubusercontent.com/1922453/30839189-b704d8c2-a2cc-11e7-84b8-e76ca9fd87f1.png">

### After
![nav](https://user-images.githubusercontent.com/1922453/30839194-c058433c-a2cc-11e7-9ff9-a2c697790ad3.gif)

folletto commented:
Would be nice to have another pair of eyes on the design, but apart from that, seems good to me 👍  

I've only a small request: can we add some left+right margin, so small screens don't have the pill touching the borders?

Current:

![screen shot 2017-09-26 at 13 51 59](https://user-images.githubusercontent.com/4389/30861194-e3b5d49e-a2c1-11e7-8608-74bc379affde.png)

8px left/right margin:

![screen shot 2017-09-26 at 13 52 43](https://user-images.githubusercontent.com/4389/30861230-00774c52-a2c2-11e7-8bea-270c7e6220c2.png)

greenafrican commented on client/blocks/stats-navigation/style.scss:
```diff
@@ -0,0 +1,51 @@
+	.section-nav__panel {
+		padding-right: 16px;
+
+		@include breakpoint( "<480px" ) {
```
Just a note that there is a preference for mobile first media queries: https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md#media-queries

greenafrican commented on client/extensions/woocommerce/app/store-stats/index.js:
```diff
@@ -64,10 +61,14 @@ class StoreStats extends Component {
+					selectedItem={ 'store' }
+					siteId={ siteId }
+					slug={ slug }
+					interval={ unit }
```
Sort alphabetically.

greenafrican commented on client/blocks/stats-navigation/style.scss:
```diff
@@ -0,0 +1,51 @@
+
+		@include breakpoint( "<480px" ) {
+			max-width: 100%;
+			margin: 0 0 9px 0;
```
Suggested from @folletto 
```
    margin: 0 8px 9px;
```

psealock commented on client/blocks/stats-navigation/index.js:
```diff
@@ -0,0 +1,97 @@
+		}
+	}
+	const { label, showIntervals, path } = navItems[ selectedItem ];
+	const slugPath = `${ slug ? '/' : '' }${ slug || '' }`;
```
lol, nice one

psealock commented on client/blocks/stats-navigation/index.js:
```diff
@@ -0,0 +1,97 @@
+	return (
+		<div className="stats-navigation">
+			<SectionNav selectedText={ label }>
+				{ isJetpack && <QueryJetpackPlugins siteIds={ jetPackSites.map( site => site.ID ) } /> }
```
Great question, I had to do some digging for this one. When switching sites, the base path is taken from the current path. So switching sites keeps the same part of Calypso visible.

But that won't work when switching to from a store to non-store site. In order to know if a newly selected site should redirect to Site Stats instead, we'll need to know about Jetpack plugins for all sites.

https://github.com/Automattic/wp-calypso/blob/1ff6fe48a4361b409cbd04a0a9bb1e170920caa0/client/components/site-selector/index.jsx#L473-L478

sirreal replied:
I tried using `siteIds={ [ siteId ] }` and saw what appeared to be the same behavior. We can save `jetpackSites - 1` requests, but there may be more complexity here than I'm appreciating.

I tried clearing mystate, refreshing on my Jetpack store site, then navigating to another Jetpack site with no store. I was redirected to `/stats/day`.

Here's the navigation with 1 site request (after clearing application state):

![nav-store-nostore](https://user-images.githubusercontent.com/841763/30960257-ca921cfc-a442-11e7-8d36-f8a1eb2063ce.gif)

These gifs aren't so great, but one is all sites, which requests plugins for all sites when you navigate between sites. The other requests 1 site for each navigation between sites.

![reqs-allsites](https://user-images.githubusercontent.com/841763/30960464-5bddc8e6-a443-11e7-8674-b10455b65fd5.gif)

![reqs-1site](https://user-images.githubusercontent.com/841763/30960467-5dda36e8-a443-11e7-87f0-542c40b1bdd1.gif)

There may be nuances or complexity here I don't appreciate, so do as you see fit 🙂

psealock replied:
I also used `siteIds={ [ siteId ] }` to confirm. The unexpected behaviour happens when you switch from one store to another. Since only the first store's plugins have been queried, it won't know the second store has Woo installed and will take the user to Site Stats.

Its a bit edge case.

sirreal replied:
Right! I get it now. 👍 

Thanks for explaining 🙂

psealock commented on client/blocks/stats-navigation/index.js:
```diff
@@ -0,0 +1,97 @@
+					className="stats-navigation__intervals-standalone"
+					selected={ interval }
+					pathTemplate={ pathTemplate }
+					standalone
```
The `Interval` component is used independently elsewhere in the app so I wanted to differentiate, but I think you are right here - theres a simpler way.

psealock commented:
Thanks for the thoughtful reviews. I'll make the appropriate internal changes and place the 8px padding on mobile.

psealock commented on client/blocks/stats-navigation/index.js:
```diff
@@ -0,0 +1,97 @@
+						.filter( isValidItem )
+						.map( item => {
+							const navItem = navItems[ item ];
+							const defaultInterval = interval || 'day';
```
On second thought, I'd like to leave it as is. This way, if the interval is ever somehow omitted from the path, the segmented control won't show a selected interval, but construction of the links to "Traffic" and "Store" won't be broken and will default to "day".

psealock commented:
### Update
* Added 8px padding on mobile to intervals
* Mobile first media queries
* Removed extraneous className on Interval component
* renamed "interval" `->` "intervalConstants"
* made `index.js` a component
* alphebetized props

Code review changes are relatively minor so I'm going to give this one a run through and merge.
